Modules/FindCUDA.cmake

```diff
@@ -734,7 +764,9 @@ if(NOT CUDA_VERSION VERSION_LESS "5.5")
   cuda_find_library_local_first(CUDA_cudart_static_LIBRARY cudart_static "static CUDA runtime library")
   mark_as_advanced(CUDA_cudart_static_LIBRARY)
 endif()
-if(CUDA_cudart_static_LIBRARY)
+
+
+if(CUDA_USE_STATIC_CUDA_RUNTIME AND CUDA_cudart_static_LIBRARY)
```

Comment:
Why are you checking for the value of CUDA_USE_STATIC_CUDA_RUNTIME here?  If someone set this with -D from the commandline or set it via option() before calling this, this check shouldn't be needed.